macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/internal/DefaultPlantUMLConfiguration.java

```diff
@@ -56,4 +56,21 @@ public String getPlantUMLServerURL()
         }
         return serverURL;
     }
+
+    @Override
+    public String getOutputFormat() {
+        String format = this.plantUMLConfigurationSource.getProperty("format");
+        // The returned value can be null if no xobject has been defined on the wiki config page.
+        if (format == null) {
+            // Fallback to xwiki.properties
+            format = this.xwikiPropertiesConfigurationSource.getProperty("plantuml.format");
+        }
+
+        if (format == null) {
+            // Fallback to default
+            format = "png";
+        }
```

Comment:
I don't think this part is needed as you could use `this.xwikiPropertiesConfigurationSource.getProperty("plantuml.format", "png");`which will use the default value if not found in `xwiki.properties`.